### PR TITLE
feat(gevent): warn when ddtrace is imported before monkey patching is done

### PR DESCRIPTION
**Pull request by jd**
This moves the profiler-only check to be for the whole ddtrace module.

### CODE REVIEW

**Review by P403n1x87 — APPROVED** [review]

File: tests/profiling/test_profiler.py [hunk 1]
```diff
@@ -258,7 +258,7 @@ def test_gevent_warning(monkeypatch):
     )
     assert subp.wait() == 0
     assert subp.stdout.read() == b""
-    assert b"RuntimeWarning: Starting the profiler before using gevent" in subp.stderr.read()
+    assert b"RuntimeWarning: Loading ddtrace before using gevent monkey patching" in subp.stderr.read()
```
Comment (brettlangdon) [inline tests/profiling/test_profiler.py:261]: should this test case be moved outside of profiling tests?

**Review by brettlangdon — COMMENTED** [review]
any updates to our docs we should have?

this note mentions ddtrace-run, should we add the same note/change that you have here about importing ddtrace after calling patch all?

https://ddtrace.readthedocs.io/en/stable/integrations.html#gevent

File: tests/profiling/test_profiler.py (hunk 1 above)
Comment (jd) [reply tests/profiling/test_profiler.py:261]: I thought about it but was not sure where to put it. We don't have a place where we run the internal/common stuff. It's all in `tracer`. Should I move it here?

Comment (brettlangdon) [reply tests/profiling/test_profiler.py:261]: yeah, that is probably where I would put it `tests/tracer` is where a lot of the internal/common tests go.

File: ddtrace/contrib/gevent/__init__.py [hunk 2]
```diff
@@ -5,10 +5,9 @@
 The integration patches the gevent internals to add context management logic.
 
 .. note::
-    If you are using :ref:`ddtrace-run<ddtracerun>` and experiencing issues with
-    patching then set ``DD_GEVENT_PATCH_ALL=true`` and ``ddtrace`` will call
-    ``gevent.monkey.patch_all()`` as early as possible in your application to
-    avoid patching conflicts.
+    If you are using :ref:`ddtrace-run<ddtracerun>` you must set ``DD_GEVENT_PATCH_ALL=true`` and ``ddtrace-run`` will
+    call ``gevent.monkey.patch_all()`` as early as possible in your application to avoid patching conflicts.
+    If you are doing manual instrumentation, you must call ``gevent.monkey.patch_all`` before importing ``ddtrace``.
```
Comment (brettlangdon) [inline ddtrace/contrib/gevent/__init__.py:10]: 👍🏻 thanks!

**Review by brettlangdon — COMMENTED** [review]
other than moving that test somewhere more common, lgtm

**Review by brettlangdon — APPROVED** [review]

**Review by Kyle-Verhoog — DISMISSED** [review]
great! :+1:

**Review by brettlangdon — DISMISSED** [review]

**Review by majorgreys — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**